### PR TITLE
[profiles] Raising an error if any profile sections are duplicated

### LINKED ISSUE #16448: [bug] profile include() does not work with os=Linux
### Describe the bug

(conan 2.4.0, on linux)

I have a base profile, `base-linux`:
```
[settings]
arch=x86_64
os=Linux
```

I have another profile, `base-clang`:
```
include(base-linux)
[settings]
compiler=clang
compiler.libcxx=libstdc++11
compiler.version=18
compiler.cppstd=20
```

And I have the top level profile, `reldeb-clang`:
```
include(base-clang-18)

[settings]
build_type=RelWithDebInfo
```

When I build with this as the host profile, the resulting output is:
```
======== Input profiles ========
Profile host:
[settings]
build_type=RelWithDebInfo
compiler=clang
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=18
```

Note, there is no os (and arch) listed, and the recipes complain and fail to build.

If I move the base-linux contents into base-clang, then it works.

There seems to be no documentation on the website about this behaviour (that I could see),
and seems to be a bug.
If the user can't do this, then conan should detect this problem and fail earlier with a clear error message.

### How to reproduce it

_No response_

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Feature: Raising a ConanException if any section is duplicated in the same Conan profile file.
Docs: omit
Closes: https://github.com/conan-io/conan/issues/16448

### CODE REVIEW

**Comment by paulharris** [comment]
Update to test is here, I assume you would have to pull it into your branch...
https://github.com/paulharris/conan/tree/issue_16448

File: conans/client/profile_loader.py [hunk 1]
```diff
@@ -263,7 +263,10 @@ def get_profile(profile_text, base_profile=None):
                                                          "options", "conf", "buildenv", "runenv",
                                                          "replace_requires", "replace_tool_requires",
                                                          "runner"])
-
+        if doc.duplicated_sections:
+            sections = ', '.join(sorted([f'[{s}]' for s in doc.duplicated_sections]))
+            ConanOutput().warning(f"Profile file has duplicated sections: {sections}. It could cause"
```
Comment (AbrilRBS) [inline conans/client/profile_loader.py]: Isnt this more of an actual error? I guess it would be breaking to raise :/

Comment (paulharris) [reply conans/client/profile_loader.py]: I would prefer to see it as an error, as conan is ignoring some settings in the shadowed section.  That could be causing some extremely weird behaviour that would be hard to detect until much later.

Comment (franramirez688) [reply conans/client/profile_loader.py]: That's why I did not raise an error. I was afraid of breaking anything. IMO, a warning is enough to realize any unexpected mistake.

Comment (franramirez688) [reply conans/client/profile_loader.py]: @memsharded wdyt?

Comment (memsharded) [reply conans/client/profile_loader.py]: I think warnings are often skipped and not seen at all.

As the behavior is basically destructive, adding a repeated section will destroy previous definition, this is a very bad thing, so I would raise an error if this condition is detected.

**Review by memsharded — APPROVED** [review]

File: conans/util/config_parser.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ def __init__(self, text, allowed_fields=None, strip_comments=False):
                 if self._allowed_fields and field not in self._allowed_fields:
                     raise ConanException("ConfigParser: Unrecognized field '%s'" % field)
                 current_lines = []
+                # Duplicated section
+                if field in self._sections:
+                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]. It could"
+                                         f"cause some unexpected behavior.")
```
Comment (AbrilRBS) [inline conans/util/config_parser.py]: Not anymore right? It could not cause new unexpected behaviour as we are disallowing it :)

Comment (franramirez688) [inline conans/util/config_parser.py]:
```suggestion
                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]")
```

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**